### ISSUE TITLE
`@analogjs/vitest-angular` doesn't run a11y tests

### ISSUE THREAD

**Issue by artaommahe** [analogjs/analog#1864]
### Please provide the environment you discovered this bug in.

- use `https://github.com/brandonroberts/angular-vite-storybook`
- add a11y addon
  - `npm add -D @storybook/addon-a11y`
  - add `'@storybook/addon-a11y'` to `addons` in `.storybook/main.ts`
  - update `.storybook/vitest.setup.ts` (https://github.com/storybookjs/storybook/issues/31700#issuecomment-2962462302)
    ```ts
    import * as addonA11y from '@storybook/addon-a11y/preview';

    setProjectAnnotations([projectAnnotations, addonA11y])
    ```
- set primary button's color `.storybook-button--primary` to `#222222` in `src/stories/button.css`
- `npx vitest`
- see no a11y errors
- open storybook and see color contrast violation for primary button in a11y tab
- run a11y tests in storybook ui and see no errors

### Which area/package is the issue in?

vitest-angular

### Description

I haven't debuged this too deep, but for me it looks like `afterEach` [function](https://github.com/storybookjs/storybook/blob/069516a18d74f3a31304fb06f66d91616d30b59d/code/addons/a11y/src/preview.tsx#L11) from `@storybook/addon-a11y/preview` is not called with this setup.
Just in case I've checked the minimum setup with clean react + vite and that function is called there and a11y errors are successfully reported both in cli and in storybook ui.

### Please provide the exception or error you saw

```true

```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [x] No
